lib/python/rose/apps/rose_bunch.py

```diff
@@ -0,0 +1,458 @@
+        if max_procs:
+            self.MAX_PROCS = int(max_procs)
+        else:
+            self.MAX_PROCS = arglength
+
+
+        if self.INCREMENTAL:
+            self.dao = RoseBunchDAO(conf_tree)
+        else:
+            self.dao = None
+
+        commands = {}
+        for index, name in enumerate(self.invocation_names):
+            invocation = RoseBunchCmd(name, self.command_format, index)
+            for key, vals in sorted(multi_args.items()):
+                invocation.argsdict[key] = shlex.split(vals.value)[index]
+            if instances:
+                invocation.argsdict["instances"] = instances[index]
+            commands[name] = invocation
+
+        procs = {}
+        log_format = os.path.join(os.environ['ROSE_TASK_LOG_DIR'], "%s")
+        #log_stdout = log_prefix + ".out"
+        #log_stderr = log_prefix + ".err"
+
+        failed = {}
+        abort = False
+
+        while procs or (commands and not abort):
+            for key, proc in procs.items():
+                if proc.poll() is not None:
+                    procs.pop(key)
+                    if proc.returncode:
+                        failed[key] = proc.returncode
+                        app_runner.handle_event(
+                                        FailEvent(key, proc.returncode))
+                        if self.dao:
+                            self.dao.update_command_state(key, self.dao.S_FAIL)
+                        if self.FAIL_HANDLE == self.TYPE_ABORT_ON_FAIL:
+                            abort = True
+                            app_runner.handle_event(AbortEvent())
+                    else:
+                        app_runner.handle_event(
+                                    SucceededEvent(key, proc.returncode))
+                        if self.dao:
+                            self.dao.update_command_state(key, self.dao.S_PASS)
+
+            while len(procs) < self.MAX_PROCS and commands and not abort:
+                key = self.invocation_names[0]
+                command = commands.pop(key)
+                self.invocation_names.pop(0)
+                cmd = command.get_command()
+                cmd_stdout = log_format % command.get_out_file()
+                cmd_stderr = log_format % command.get_err_file()
+
+                if self.dao:
+                    if self.dao.check_has_succeeded(key):
+                        app_runner.handle_event(SkippingEvent(key))
+                        continue
+                    else:
+                        self.dao.add_command(key)
+
+                app_runner.handle_event(LaunchEvent(key, cmd))
+                procs[key] = app_runner.popen.run_bg(
+                                                cmd,
+                                                shell=True,
+                                                stdout=open(cmd_stdout, 'w'),
+                                                stderr=open(cmd_stderr, 'w'))
+            sleep(self.SLEEP_DURATION)
+
```

Comment:
Given that you have redirected the STDOUT and STDERR of individual commands to alternate locations, it is probably worth using the STDOUT of the built-in application to report the status of the commands. In this case, a status line to indicate that a command was never started should be sufficient.